### PR TITLE
Add --vae_precision option to the SDXL pix2pix script so that we have…

### LINKED ISSUE #4424: [SD-XL] Instruct Pix2Pix fine-tuning script unconditionally upcasts VAE to float32 when built-in VAE is used
```py
    # Move vae, unet and text_encoder to device and cast to weight_dtype
    # The VAE is in float32 to avoid NaN losses.
    if args.pretrained_vae_model_name_or_path is not None:
        vae.to(accelerator.device, dtype=weight_dtype)
    else:
        vae.to(accelerator.device, dtype=torch.float32)
```

I believe this causes the training to fail when operating in mixed precision, eg. `fp16` or `bf16`:

```
RuntimeError: Input type (c10::BFloat16) and bias type (float) should be the same
```

### PR DESCRIPTION
**Pull request by bghira**
… the option of avoiding float32 overhead

# What does this PR do?

Fixes #4424 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: examples/instruct_pix2pix/train_instruct_pix2pix_sdxl.py [hunk 1]
```diff
@@ -878,7 +890,7 @@ def collate_fn(examples):
     if args.pretrained_vae_model_name_or_path is not None:
         vae.to(accelerator.device, dtype=weight_dtype)
```
Comment (bghira) [inline examples/instruct_pix2pix/train_instruct_pix2pix_sdxl.py:891]: wasn't sure if we want to keep the original value here or just clean it up and keep it all consistent via vae_precision

**Review by patrickvonplaten — COMMENTED** [review]
@sayakpaul wdyt?

**Review by sayakpaul — APPROVED** [review]
LGTM

**Comment by patrickvonplaten** [comment]
Let's just make sure `make style` is run here

**Comment by bghira** [comment]
@patrickvonplaten good to go

**Pull request merged**